roles/api-environment/README.md

```diff
@@ -0,0 +1,51 @@
+API Environment
+===
+
+This role sets up servers environment to run different kind of APIs within docker containers. The following are three main parts of this role.
+
+## Generic Tasks
+
+In this stage, an api user will be created if not exists yet. **DO NOT** put the user password in `defaults/mains.yml`, instead, define your user password at the playbook level. For example:
```

Comment:
I wouldn't recommend putting the password in playbook either. [Ansible mentions the password on darwin/macOS systems has to be in plain text](https://docs.ansible.com/ansible/2.5/modules/user_module.html), but sorta since the text is the [hashed version of the password](https://docs.ansible.com/ansible/faq.html#how-do-i-generate-crypted-passwords-for-the-user-module). Either way, I think this text shouldn't be in the playbook but instead be in the inventory file.

Reply:
Yes, we should be using vault, but the password for creating a user in the ansible user module is actually the hashed version of the password, so its not really plain text. I still don't think it should be in the playbook, since playbooks are meant to be shared.